File: Java/Ejercicio2.java
import java.util.ArrayList;

public class Ejercicio2 {
    public static void main(String[] args) {

        // Creamos el ArrayList para guardar las notas
        ArrayList<Double> notas = new ArrayList<>();

        // Agregamos 6 notas
        notas.add(8.5);
        notas.add(9.0);
        notas.add(7.5);
        notas.add(6.0);
        notas.add(9.5);
        notas.add(8.0);

        // Calculamos la suma de todas las notas
        double suma = 0;
        for (int i = 0; i < notas.size(); i++) {
            suma = suma + notas.get(i);
        }

        // Calculamos el promedio
        double promedio = suma / notas.size();

        // Mostramos cada nota
        System.out.println("Notas registradas:");
        for (int i = 0; i < notas.size(); i++) {
            System.out.println("Nota " + (i + 1) + ": " + notas.get(i));
        }

        // Mostramos el promedio
        System.out.println("Promedio del grupo: " + promedio);
    }
}

File: C++/ejercicio2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {

    // Creamos el vector para guardar las notas
    vector<double> notas;

    // Agregamos 6 notas
    notas.push_back(8.5);
    notas.push_back(9.0);
    notas.push_back(7.5);
    notas.push_back(6.0);
    notas.push_back(9.5);
    notas.push_back(8.0);

    // Calculamos la suma de todas las notas
    double suma = 0;
    for (int i = 0; i < notas.size(); i++) {
        suma = suma + notas[i];
    }

    // Calculamos el promedio
    double promedio = suma / notas.size();

    // Mostramos cada nota
    cout << "Notas registradas:" << endl;
    for (int i = 0; i < notas.size(); i++) {
        cout << "Nota " << (i + 1) << ": " << notas[i] << endl;
    }

    // Mostramos el promedio
    cout << "Promedio del grupo: " << promedio << endl;

    return 0;
}
